hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/ServicesStateTest.java

```diff
@@ -989,4 +1014,65 @@ private void unmockMigrators() {
         ServicesState.setStakingInfoBuilder(StakingInfoMapBuilder::buildStakingInfoMap);
         ServicesState.setVmFactory(VirtualMapFactory::new);
     }
+
+    /**
+     * Recursively copies a directory from {@code sourceDir} to targetDir, transforming both path
+     * segments and file names with the given function.
+     *
+     * @param sourceDir the source directory
+     * @param targetDir the target directory
+     * @param transform the function to apply to each path segment and file name
+     * @throws IOException if an I/O error occurs
+     */
+    private void cpWithDirTransform(
```

Comment:
Do we even need to have test that loads 0.30 state ? Can we change minimum supported version to 0.31 ?